doc/bootloader/README.md

```diff
@@ -0,0 +1,113 @@
+
+The SPI flash memory contains a recovery firmware and an area big enough to store a whole firmware (the new firmware will be first written to this area before being swapped into the internal memory). The remaining space is available for the application firmware needs.
+
+### The bootloader
+[The bootloader](https://github.com/InfiniTimeOrg/pinetime-mcuboot-bootloader) is based on [MCUBoot, a secure bootloader for 32 bits microcontrollers](https://github.com/mcu-tools/mcuboot). It was originally written by [Lup Yuen](https://github.com/lupyuen).
+
+MCUBoot is run at boot time. In normal operation, it just jumps to the reset handler of the application firmware to run it. Once the application firmware is running, MCUBoot does not run at all.
+
+![MCUBoot boot sequence diagram](boot.png)
+
+But MCUBoot does much more than that : it can upgrade the firmware that is currently running by a new one, and it is also able to revert to the previous version of the firmware in case the new one does not run properly.
+
+To do this, it uses 2 memory 'slots' :
+- **The primary slot** (located in the internal flash memory of the nRF52832) : it contains the current firmware, the one that will be executed by MCUBoot
+- **The secondary slot** (located in the external SPI flash memory) : it is used to store the upgraded version of the firmware, when available.
+
+At boot time, MCUBoot detects that a new version of the firmware is available in the secondary slot and swaps them : the current version of the firmware is copied from the primary to the secondary slot and vice-versa.
+
+When the swapping is done, it runs the new version of the firmware from the primary slot.
+
+![MCUBoot update sequence diagram](upgrade.png)
+
+The next time MCUBoot will be run (after an MCU reset), MCUBoot will check if the new firmware ran correctly (it must set a flag in memory). If it is not the case, it'll revert to the previous version of the firmware by copying it from the secondary to the primary.
+
+![MCUBoot update sequence diagram](recover.png)
+
+You'll find more information about MCUBoot, its design and how it works in [the documentation of the project](https://docs.mcuboot.com/).
+
+The [pinetime-mcuboot-bootloader](https://github.com/InfiniTimeOrg/pinetime-mcuboot-bootloader) integrates MCUBoot, provides low-level drivers for the display and SPI flash memory and implements a very basic UI. See [the documentation of the project for more details about this](https://github.com/InfiniTimeOrg/pinetime-mcuboot-bootloader#boot-flow).
+
+An application firmware must implement the following functionalities to be compatible with our bootloader:
+
+- The application firmware will be flashed at **address 0x8000 of the internal flash**. It must then be **linked** with a start address of 0x8000 (instead of 0x00 by default).
```

Comment:
```suggestion
- The application firmware will be flashed at **address 0x8000 of the internal flash**.
  It must then be **linked** with a start address of 0x8000 (instead of 0x00 by default).
```

Reply:
Since this is a list of requirements for the application firmware, perhaps list the requirement itself first, then state the rationale. eg:
```suggestion
- It must be **linked** with a start address of 0x8000 (instead of the default of 0x00).
  This is because the first 32KB of internal flash is reserved for the bootloader, so the application firmware image will be copied to the internal flash at an **offset address** of 0x8000.
```